### PR TITLE
feat: update and add new core-v3 endpoints

### PR DESCRIPTION
**Pull request by KovacZan**
## Summary
Added and updated core-v3 endpoints.
Made it possible to work with dot notation.
Had to adjust parameters like this:
```
"timestamp.from"?: number;
"timestamp.to"?: number;
```
instead of:
```
timestamp?: {
    from?: number;
    to?: number;
}
```
because GET method doesn't parse the object to dot notation.

## Checklist

- [ ] Documentation _(if necessary)_
- [ ] Tests _(if necessary)_
- [x] Ready to be merged

### CODE REVIEW

File: src/resources/blockchain.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Resource } from "./resource";
+import { ApiResponse } from "../interfaces";
+import { Blockchain as BlockchainResource } from "../resourcesTypes";
+/**
+ * Retrieve Blockchain State
+ */
+export class Blockchain extends Resource {
+	/**
+	 * Used to get the latest block and supply of the blockchain.
```
Comment (faustbrian) [inline src/resources/blockchain.ts:9]: Remove

File: src/resources/blockchain.ts [hunk 2]
```diff
@@ -0,0 +1,14 @@
+import { Resource } from "./resource";
+import { ApiResponse } from "../interfaces";
+import { Blockchain as BlockchainResource } from "../resourcesTypes";
+/**
+ * Retrieve Blockchain State
```
Comment (faustbrian) [inline src/resources/blockchain.ts:5]: Remove

**Review by faustbrian — CHANGES_REQUESTED** [review]
The input shouldn't change, it should simply be transformed to dot-notation before being sent to the API.

**Review by faustbrian — CHANGES_REQUESTED** [review]
Use https://github.com/sindresorhus/dot-prop instead of the current implementation.

**Comment by faustbrian** [comment]
Closing this PR because it contains a large amount of unrelated changes. Feel free to reopen smaller PRs that are submitting specific changes.

**Comment by KovacZan** [comment]
Could you open a specific branch for core-v3 changes.

**Pull request closed without merging**